src/components/transactions/TxDetails/index.tsx

```diff
@@ -55,6 +57,14 @@ const TxDetailsBlock = ({ txSummary, txDetails }: TxDetailsProps): ReactElement
   // If we have no token list we always trust the transfer
   const isTrustedTransfer = !hasDefaultTokenlist || isTrustedTx(txSummary)
 
+  const proposedSafeTxHash =
+    isMultisigDetailedExecutionInfo(txDetails.detailedExecutionInfo) &&
+    isMultisigExecutionInfo(txSummary.executionInfo) &&
+    wallet &&
+    wallet.address === txSummary.executionInfo.proposer?.value
```

Comment:
Let's use the `sameAddress` helper here

Reply:
A test for this would be good as well as this code decides if we offer deletion for the owner or not.